packages/MJServer/src/generic/ResolverBase.ts

```diff
@@ -818,4 +818,25 @@ export class ResolverBase {
     return true;
   }
   protected async AfterUpdate(dataSource: DataSource, input: any) {}
+
+  protected GetReadOnlyDataSource(dataSources: DataSourceInfo[], allowFallbackToReadWrite: boolean): DataSource {
```

Comment:
Having an arg for `allowFallbackToReadWrite` is a great idea, although writing it as a positional argument isn't great since when you read a call to this elsewhere in the code it looks like `this.GetReadOnlyDataSource(dataSources, true);` and the reader is left to wonder wtf `true` is all about. A more readable (dev-friendly) approach would be to use an options argument (which is a common JS pattern to gain named args):

```ts
type GetDataSourceOptions = { allowFallbackToReadWrite?: boolean };

protected GetReadOnlyDataSource(dataSources: DataSourceInfo[], options?: GetDataSourceOptions): DataSource {
  const { allowFallbackToReadWrite } = options ?? {};
```

Then when calling it, the meaning is clear: `this.GetReadOnlyDataSource(dataSource, { allowFallbackToReadWrite: true });`